hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/RatisPipelineProvider.java

```diff
@@ -147,10 +156,15 @@ public synchronized Pipeline create(RatisReplicationConfig replicationConfig,
       List<DatanodeDetails> excludedNodes, List<DatanodeDetails> favoredNodes)
       throws IOException {
     if (exceedPipelineNumberLimit(replicationConfig)) {
-      throw new SCMException("Ratis pipeline number meets the limit: " +
-          pipelineNumberLimit + " replicationConfig : " +
-          replicationConfig,
-          SCMException.ResultCodes.FAILED_TO_FIND_SUITABLE_NODE);
+      String limitInfo = (maxPipelinePerDatanode > 0)
+          ? String.format("per datanode: %d", maxPipelinePerDatanode)
+          : String.format(": %d", pipelineNumberLimit);
+
+      throw new SCMException(
+          String.format("Ratis pipeline number meets the limit %s replicationConfig: %s",
```

Comment:
I think "Cannot create pipeline as it would exceed the limit" would be easier to understand than "Ratis pipeline number meets the limit".